icu4c/source/i18n/messageformat2_formattable.cpp

```diff
@@ -259,9 +255,9 @@ namespace message2 {
         switch (type) {
         case UFMT_DATE: {
             UnicodeString result;
-            UDate d = toFormat.getDate(status);
+            const DateInfo* dateInfo = toFormat.getDate(status);
             U_ASSERT(U_SUCCESS(status));
-            formatDateWithDefaults(locale, d, result, status);
+            formatDateWithDefaults(locale, *dateInfo, result, status);
```

Comment:
dateInfo could be nullptr here, you cannot just deref here. Also, status could be error

Reply:
There's an assertion right before, and `getDate()` returns a non-null pointer if it succeeds, but I changed the code and added a comment to make this clearer.